test/e2e/delete_test.go

```diff
@@ -35,25 +35,35 @@ func testWindowsNodeDeletion(t *testing.T) {
 	}
 
 	require.NotNil(t, windowsMachineSetWithLabel, "could not find MachineSet with Windows label")
-	// Set the expected number of Windows nodes to 0
-	gc.numberOfNodes = 0
```

Comment:
What if we add further tests where this number needs to be accurate? Figuring that it needed to be reset here is going to be hard.

Reply:
I'm not fully understanding the case you are describing. My issue is that throughout all the other tests gc.numberOfNodes is set through test arguments. If this needs to be changed in the deletion tests I feel it should be passed in when initiating the tests. Otherwise, normal non-global variables should be used.